File: pycurses/visettings.py
# -*- coding: utf-8 -*-
"""
Created on Sun Apr 11 17:37:37 2010

@author: jacob
"""

class Visettings:
    __shared_state = {}
    Screen = 0
    
    CurrentMode = 0
    Speed = 0
    PlaySpeed = 100
    
    FrameNumber = 0
    
    MaxX = 0
    MaxY = 0
    
    Player1Name = 0
    Player2Name = 0
    
    Player1Plants = 0
    Player2Plants = 0
    
    AutoEnd = 0
    
    CurX = 0
    CurY = 0
    
    Done = 0
    FirstRun = 1
    
    P1TextF = 1
    P1TextB = 0
    
    P2TextF = 4
    P2TextB = 0
    
    P1PieceF = 7
    P1PieceB = 1
    
    P2PieceF = 7
    P2PieceB = 4
    
    InstrF = 7
    InstrB = 0
    
    BackgroundF = 2
    BackgroundB = 0
    
    DefaultF = 7
    DefaultB = 0
    
    SelectionF = 0
    SelectionB = 6
    
    Game = 0
    State = 0
    
    
    def __init__(self):
        self.__dict__ = self.__shared_state
    
    
    

File: ncurses/visettings.cpp
#include "visettings.h"

visettings *visettings::inst = NULL;

visettings::visettings()
{
  winAnimation = true;
}

visettings::~visettings()
{
  if( inst )
    delete inst;
  inst = 0;
}

void visettings::loadFromFile()
{
  Config configFile( "VISCONFIG" );

  setAttr( winAnimation, configFile.pBool( "winAnimation" ) );

  setAttr( p1pieceF, configFile.pInt( "P1PieceForeground" ) );
  setAttr( p1pieceB, configFile.pInt( "P1PieceBackground" ) );

  setAttr( p1textF, configFile.pInt( "P1TextForeground" ) );
  setAttr( p1textB, configFile.pInt( "P1TextBackground" ) );

  setAttr( p2pieceF, configFile.pInt( "P2PieceForeground" ) );
  setAttr( p2pieceB, configFile.pInt( "P2PieceBackground" ) );

  setAttr( p2textF, configFile.pInt( "P2TextForeground" ) );
  setAttr( p2textB, configFile.pInt( "P2TextBackground" ) );

  setAttr( instrF, configFile.pInt( "InstructionsForeground" ) );
  setAttr( instrB, configFile.pInt( "InstructionsBackground" ) );

  setAttr( backgroundF, configFile.pInt( "BackgroundForeground" ) );
  setAttr( backgroundB, configFile.pInt( "BackgroundBackground" ) );

  setAttr( selectionF, configFile.pInt( "SelectionForeground" ) );
  setAttr( selectionB, configFile.pInt( "SelectionBackground" ) );

  setAttr( defaultF, configFile.pInt( "DefaultForeground" ) );
  setAttr( defaultB, configFile.pInt( "DefaultBackground" ) );

  setAttr( alwaysWin, configFile.pBool( "AlwaysPlayWin" ) );

  setAttr( playSpeed, configFile.pInt( "DefaultSpeed" ) );

  setAttr( autoEnd, false );
  setAttr( done, false );

  setAttr( startPause, configFile.pInt( "STARTPAUSE" ) );
  setAttr( endPause, configFile.pInt( "ENDPAUSE" ) );
}
